src/Common/Core/Impl/Disposables/DisposeToken.cs

```diff
@@ -14,7 +14,7 @@ public sealed class DisposeToken {
             return new DisposeToken(Invariant($"{typeof(T).Name} instance is disposed"));
         }
 
-        public bool IsDisposed => _disposed == 0;
+        public bool IsDisposed => _disposed == 1;
```

Comment:
While we are at it - in order to unblock tests, shall we at least apply a workaround in `DisconnectAsync` - only `await _runTask` when it is running (or do we actually care at all?) and same in `StopHostAsync`. Also, check disposed in `RSessionOnDisconnectedAsync` and not call `TestBrokerConnectionWithRHost`.